packaging/installer/kickstart-ng.sh

```diff
@@ -0,0 +1,842 @@
+      run ${ROOTCMD} ${pm_cmd} env ${env} ${uninstall_subcmd} ${pkg_install_opts} "${repoconfig_name}"
+    fi
+    return 2
+  fi
+
+  progress "Installing Netdata package."
+  # shellcheck disable=SC2086
+  if ! run ${ROOTCMD} env ${env} ${pm_cmd} install ${pkg_install_opts} netdata; then
+    warning "Failed to install Netdata package."
+    if [ -z "${NO_CLEANUP}" ]; then
+      progress "Attempting to uninstall repository configuration package."
+      # shellcheck disable=SC2086
+      run ${ROOTCMD} env ${env} ${pm_cmd} ${uninstall_subcmd} ${pkg_install_opts} "${repoconfig_name}"
+    fi
+    return 2
+  fi
+}
+
+# ======================================================================
+# Static build install code
+
+set_static_archive_urls() {
+  if [ "${RELEASE_CHANNEL}" = "stable" ]; then
+    latest="$(download "https://api.github.com/repos/netdata/netdata/releases/latest" /dev/stdout | grep tag_name | cut -d'"' -f4)"
+    export NETDATA_STATIC_ARCHIVE_URL="https://github.com/netdata/netdata/releases/download/${latest}/netdata-${SYSARCH}-${latest}.gz.run"
+    export NETDATA_STATIC_ARCHIVE_CHECKSUM_URL="https://github.com/netdata/netdata/releases/download/${latest}/sha256sums.txt"
+  else
+    export NETDATA_STATIC_ARCHIVE_URL="${NETDATA_TARBALL_BASEURL}/netdata-latest.gz.run"
+    export NETDATA_STATIC_ARCHIVE_CHECKSUM_URL="${NETDATA_TARBALL_BASEURL}/sha256sums.txt"
+  fi
+}
+
+try_static_install() {
+  set_static_archive_urls "${RELEASE_CHANNEL}"
+  progress "Downloading static netdata binary: ${NETDATA_STATIC_ARCHIVE_URL}"
+
+  if ! download "${NETDATA_STATIC_ARCHIVE_URL}" "${tmpdir}/netdata-${SYSARCH}-latest.gz.run"; then
+    warning "Unable to download static build archive for ${SYSARCH}."
+    return 2
+  fi
+
+  if ! download "${NETDATA_STATIC_ARCHIVE_CHECKSUM_URL}" "${tmpdir}/sha256sum.txt"; then
+    fatal "Unable to fetch checksums to verify static build archive."
+  fi
+
+  if ! grep "netdata-${SYSARCH}-latest.gz.run" "${tmpdir}/sha256sum.txt" | safe_sha256sum -c - > /dev/null 2>&1; then
+    fatal "Static binary checksum validation failed. Stopping Netdata Agent installation and leaving binary in ${tmpdir}. Usually this is a result of an older copy of the file being cached somewhere upstream and can be resolved by retrying in an hour."
+  fi
+
+  if [ "${INTERACTIVE}" -eq 0 ]; then
+    opts="${opts} --accept"
+  fi
+
+  progress "Installing netdata"
+  # shellcheck disable=SC2086
+  if ! run ${ROOTCMD} sh "${tmpdir}/netdata-${SYSARCH}-latest.gz.run" ${opts} -- ${NETDATA_AUTO_UPDATES:+--auto-update} ${NETDATA_INSTALLER_OPTIONS}; then
+    fatal "Failed to install static build of Netdata on ${SYSARCH}."
+  fi
+
+  install_type_file="/opt/netdata/etc/netdata/.install-type"
+  if [ -f "${install_type_file}" ]; then
+    # shellcheck disable=SC1090
+    . "${install_type_file}"
+    ${ROOTCMD} cat > "${install_type_file}" <<- EOF
+	INSTALL_TYPE='kickstart-static'
+	PREBUILT_ARCH='${PREBUILT_ARCH}'
+	EOF
+  fi
```

Comment:
This could could have a problem, and while testing in Gentoo (which will install the static package. I run the script as user.) I got:

```
./kickstart-ng.sh: line 689: /opt/netdata/etc/netdata/.install-type: Permission denied
 ABORTED  Unable to install on this system.
```

(It's trying to open the file as a local user, before executing sudo).
What I found to work is maybe using tee:

```
        $ROOTCMD tee "${install_type_file}" > /dev/null <<-EOF
INSTALL_TYPE='kickstart-static'
PREBUILT_ARCH='${PREBUILT_ARCH}'
EOF
```

And also, removed tabs that might be inserted in the file. 

We should also check that the permissions of this file are ok (they should be since it is created from the static tarball) but we should keep in mind...

Reply:
Yeah, this is one of two places we have an issue with this. Working on a fix right now.

Reply:
Should be fixed now.

Reply:
Sorry to have to re-open it, just tried, and got:

```
./kickstart-ng.sh: line 707: /opt/netdata/etc/netdata/.install-type: Permission denied
 ABORTED  Unable to install on this system.
```

Reply:
Updated again, should actually be fixed this time (I forgot to fix writing out the updated install type file).

Reply:
Yes, looks ok! But there is a small typo in 707, there is an extra `}` character after `install-type}`